api/test/infrastructure/google/google.client.test.ts

```diff
@@ -1,49 +1,144 @@
-        redirectPath: '/api/v1/google/sign-in/token',
-        authEndpoint: 'https://accounts.google.com/o/oauth2/auth',
-        tokenEndpoint: 'https://oauth2.googleapis.com/token'
-      }
-    };
     client = new GoogleClient(config);
   });
 
   beforeEach(() => {
     jest.clearAllMocks();
   });
 
-  it('should build valid OIDC request', () => {
-    const givenBaseUrl = 'http://localhost:8080';
-    const givenState = randomUUID();
-    const expectedQueyObject = {
-      client_id: config.oauth.clientId,
-      response_type: 'code',
-      redirect_uri: `${givenBaseUrl}${config.oauth.redirectPath}`,
-      scope: 'openid email',
-      state: givenState,
-      access_type: 'online',
-      prompt: 'consent select_account'
+  describe('Test build OIDC request', () => {
+    it('should build valid OIDC request', () => {
+      const givenState = randomUUID();
+      const expectedQueyObject = {
+        client_id: config.oauth.clientId,
+        response_type: 'code',
+        redirect_uri: `${baseUrl}${config.oauth.redirectPath}`,
+        scope: 'openid email',
+        state: givenState,
+        access_type: 'online',
+        prompt: 'consent select_account'
+      };
+
+      const result = client.buildOidcRequest(baseUrl, givenState);
+      const [authEndpoint, queryString] = result.split('?');
+
+      expect(authEndpoint).toEqual(config.oauth.authEndpoint);
+      expect(parse(queryString)).toEqual(
+        expect.objectContaining(expectedQueyObject)
+      );
+    });
+  });
+
+  describe('Test exchange auth code', () => {
+    const authCode = 'randomAuthCode';
+    const requestConfig = {
+      headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
+    };
+    const response = {
+      status: 200,
+      data: {
+        access_token: 'access_token',
+        expires_in: 3600,
+        scope: 'https://www.googleapis.com/auth/userinfo.email openid',
+        token_type: 'Bearer',
+        id_token: 'id_token'
+      },
+      getHttpStatusCode: jest.fn(() => {
+        return 200;
+      })
     };
+    const encodedBody = new URLSearchParams({
+      code: authCode,
+      client_id: config.oauth.clientId,
+      client_secret: config.oauth.clientSecret,
+      redirect_uri: `${baseUrl}${config.oauth.redirectPath}`,
+      grant_type: 'authorization_code'
+    });
+
+    it('should success when valid', async () => {
+      response.status = 200;
+      response.getHttpStatusCode = jest.fn(() => {
+        return 200;
+      });
+      jest.spyOn(axios, 'post').mockResolvedValue(response);
+
+      const result = await client.exchangeAuthCode(baseUrl, authCode);
+
+      expect(result).toEqual(response.data.id_token);
+      expect(axios.post).toBeCalledTimes(1);
+      expect(axios.post).toBeCalledWith(
+        config.oauth.tokenEndpoint,
+        encodedBody,
+        requestConfig
+      );
+    });
+
+    it('should failure when token API request fails', async () => {
```

Comment:
```suggestion
    it('should fail when token API request fails', async () => {
```

양쪽 it 노드에 문장이 어색한 것 같아 고쳐주시면 감사하겠습니다~

Reply:
확인했습니다. https://github.com/MovieReviewComment/Mr.C/pull/105/commits/24c6911d186b7fc747e36d7cc59bfc1c69020272 에서 수정했습니다.